canvas_modules/common-canvas/src/common-canvas/cc-right-flyout.jsx

```diff
@@ -19,24 +19,73 @@ import { connect } from "react-redux";
 import PropTypes from "prop-types";
 import Logger from "../logging/canvas-logger.js";
 
+const MIN_WIDTH = 300;
+const MAX_WIDTH_EXTEND_PERCENT = 0.7; // Should cover atmost 70% of available width
 class CommonCanvasRightFlyout extends React.Component {
 	constructor(props) {
 		super(props);
+
 		this.logger = new Logger("CC-RightFlyout");
+
+		this.onMouseDown = this.onMouseDown.bind(this);
+		this.onMouseMoveX = this.onMouseMoveX.bind(this);
+		this.onMouseUp = this.onMouseUp.bind(this);
+	}
+
+	onMouseDown(e) {
+		if (e.button === 0) {
+			document.addEventListener("mousemove", this.onMouseMoveX, true);
+			document.addEventListener("mouseup", this.onMouseUp, true);
+			this.posX = e.clientX;
+			this.startWidth = this.commonCanvasRightFlyout.offsetWidth;
+
+			e.preventDefault();
+		}
+	}
+
+	onMouseUp() {
+		document.removeEventListener("mousemove", this.onMouseMoveX, true);
+		document.removeEventListener("mouseup", this.onMouseUp, true);
+	}
+
+	onMouseMoveX(e) {
+		if (e.clientX) {
+			const newWidth = this.startWidth + (this.posX - e.clientX);
+			this.props.canvasController.setRightPanelWidth(this.limitWidth(newWidth));
+		}
+	}
+
+	// Returns a new width for right panel limited by the need to enforce
+	// a minimum and maximum width
+	limitWidth(wth) {
+		const canvasContainer = document.getElementById(this.props.containingDivId);
+		let width = wth;
+
+		if (canvasContainer) {
+			// Max Width should be 70% of the total available width (canvas + rightflyout)
+			const totalAvialableWidth = canvasContainer.getBoundingClientRect().width + this.commonCanvasRightFlyout?.offsetWidth;
+			const maxWidth = MAX_WIDTH_EXTEND_PERCENT * totalAvialableWidth;
+			width = Math.min(Math.max(width, MIN_WIDTH), maxWidth);
+		}
+
+		return width;
 	}
 
 	render() {
 		this.logger.log("render");
-
```

Comment:
Just a trivial thing but ... please don't remove lines like this used for spacing unless there is a very good reason to remove it.